services/horizon/cmd/root.go

```diff
@@ -40,17 +46,50 @@ func validateBothOrNeither(option1, option2 string) {
 	}
 }
 
+func pingDB(db *sql.DB) {
+	for attempt := 0; attempt < maxDBPingAttempts; attempt++ {
+		err := db.Ping()
+		if err == nil {
+			return
+		}
+		stdLog.Printf("could not ping horizon db: %v\n", err)
+		time.Sleep(time.Second)
+		if attempt+1 < maxDBPingAttempts {
+			stdLog.Println("retrying ping")
```

Comment:
Make the message more clear. Maybe:
```suggestion
			stdLog.Println("Waiting for a DB connection...")
```